saltapi/service/proposal_service.py

```diff
@@ -189,52 +189,82 @@ async def put_proposal_progress(
-            # fmt: off
-            "description_of_observing_constraints":
-                proposal_progress_report.description_of_observing_constraints,
+            "description_of_observing_constraints": proposal_progress_report.description_of_observing_constraints,
             "change_reason": proposal_progress_report.change_reason,
-            # fmt: off
-            "summary_of_proposal_status":
-                proposal_progress_report.summary_of_proposal_status,
+            "summary_of_proposal_status": proposal_progress_report.summary_of_proposal_status,
             "strategy_changes": proposal_progress_report.strategy_changes,
             "partner_requested_percentages": partner_requested_percentages,
         }
 
-        # Get the content and filename of the additional pdf, if there is one
+    async def handle_additional_pdf(
+            self,
+            additional_pdf: Optional[UploadFile]
+    ) -> Tuple[Optional[str], Optional[bytes]]:
         additional_pdf_filename = None
         additional_pdf_content = b""
         if additional_pdf:
             additional_pdf_content = await additional_pdf.read()
-            additional_pdf_filename = (
-                self.repository.generate_proposal_progress_filename(
-                    additional_pdf_content, is_supplementary=True
-                )
+            additional_pdf_filename = self.repository.generate_proposal_progress_filename(
+                additional_pdf_content,
+                is_supplementary=True
             )
-
-        # Store the progress report in the database
-        filenames = {
-            "proposal_progress_filename": None,
-            "additional_pdf_filename": additional_pdf_filename,
-        }
-        self.repository.put_proposal_progress(
-            proposal_progress, proposal_code, semester, filenames
+        return additional_pdf_filename, additional_pdf_content
+
+    async def handle_proposal_progress_pdf(
+            self,
+            proposal_code: str,
+            semester: str
+    ) -> Tuple[str, bytes]:
+        proposal_progress_byte_io = self.generate_proposal_progress_pdf(
+            cast(ProposalCode, proposal_code),
+            cast(Semester, semester)
         )
+        progress_pdf_content = proposal_progress_byte_io.read()
+        progress_pdf_filename = self.repository.generate_proposal_progress_filename(progress_pdf_content)
+        return progress_pdf_filename,  progress_pdf_content
 
-        # Save the additional file to disk. This is done last as the file should not be
-        # stored if the progress report cannot be stored in the database.
-        if additional_pdf_filename:
-            additional_pdf_path = (
-                ProposalService._included_dir(proposal_code) / additional_pdf_filename
-            )
-            additional_pdf_path.write_bytes(additional_pdf_content)
+    @staticmethod
+    async def save_progress_report_pdf(
```

Comment:
Looking at the code, it doesn't really matter whether the file is a pdf, so you might remove `pdf` from the function name and from the variable names within the function.